tfx/components/example_gen/utils.py

```diff
@@ -241,27 +256,50 @@ def _retrieve_latest_span_version(
       - If Span or Version found is not an integer.
       - If a matching cannot be found for split pattern provided.
   """
+  is_match_span = SPAN_SPEC in split.pattern
+  is_match_date = any(spec in split.pattern for spec in DATE_SPECS)
+  if [is_match_span, is_match_date].count(True) > 1:
+    raise ValueError('Either span spec or date specs must be specified '
+                     'exclusively.')
 
   split_pattern = os.path.join(uri, split.pattern)
 
   split_glob_pattern = split_pattern
   split_regex_pattern = _glob_to_regex(split_pattern)
 
-  latest_span = None
-  latest_version = None
-
-  if SPAN_SPEC not in split.pattern:
+  if not is_match_span and not is_match_date:
     if VERSION_SPEC in split.pattern:
-      raise ValueError('Version spec provided, but Span spec is not present')
-    return latest_span, latest_version
+      raise ValueError('Version spec provided, but Span or Date spec is not '
+                       'present.')
+    return (None, None)
 
-  if split.pattern.count(SPAN_SPEC) != 1:
+  if is_match_span and split.pattern.count(SPAN_SPEC) != 1:
     raise ValueError('Only one %s is allowed in %s' %
                      (SPAN_SPEC, split.pattern))
+  elif is_match_date and not all(split.pattern.count(spec) == 1
+                                 for spec in DATE_SPECS):
+    raise ValueError('Exactly one of each date spec is required in %s' %
+                     split.pattern)
```

Comment:
nit, consider move all the validation to to some util

Reply:
fixed. moved all initial spec finding validation into `_verify_split_specs()`, which returns bools indicating what should be matched